master/internal/stream/messages.go

```diff
@@ -0,0 +1,43 @@
+package stream
+
+import (
+	"encoding/json"
+
+	"github.com/gorilla/websocket"
+	log "github.com/sirupsen/logrus"
+
+	"github.com/determined-ai/determined/master/pkg/stream"
+)
+
+// StartupMsg is the first message a streaming client sends.
+//
+// It declares initially known keys and also configures the initial subscriptions for the stream.
+type StartupMsg struct {
+	SyncID    string              `json:"sync_id"`
+	Known     KnownKeySet         `json:"known"`
+	Subscribe SubscriptionSpecSet `json:"subscribe"`
+}
+
+// KnownKeySet allows a client to describe which primary keys it knows of as existing, so the server
+// can respond with a different KnownKeySet of deleted messages of client-known keys that don't
+// exist.
```

Comment:
... so the server can respond with which client-known keys have been deleted or disappeared, and also which server-known keys are not yet known to the client (appearances).